scico/linop/_xray.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020-2023 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SCICO package. Details of the copyright and
+# user license can be found in the 'LICENSE' file distributed with the
+# package.
+
+
+"""
+X-ray projector classes.
+"""
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+
+from scico.typing import Shape
+
+from ._linop import LinearOperator
+
+
+class XRayProject(LinearOperator):
+    """X-ray projection operator.
+
+    Wraps an X-ray projector object in a SCICO
+    :class:`LinearOperator`.
+    """
+
+    def __init__(self, projector):
+        r"""
+        Args:
+            projector: instance of an X-ray projector object to wrap,
+                currently the only option is
+                :class:`ParallelFixedAxis2dProjector`
+        """
+        self._eval = projector.project
+
+        super().__init__(
+            input_shape=projector.im_shape,
+            output_shape=(len(projector.angles), *projector.det_shape),
+        )
+
+
+class ParallelFixedAxis2dProjector:
```

Comment:
Quite a few different properties squeezed into the name! If any of these may be generalized (e.g. 2d or 3d), consider moving the property out of the name and making it a parameter when the alternative is also supported.